articles/azure/azs-how-create-vm-terraform.md

```diff
@@ -125,34 +103,65 @@ Official [Variables Guide](https://www.terraform.io/intro/getting-started/variab
 
 - `tenant_id` - The tenant ID of your Azure Active Directory tenant domain. It can either be the actual GUID or your Azure Active Directory tenant domain name.
 
+> [!NOTE]
+> You can also put the content of `variables.tf`  at the top of the `main.tf` file.
+
+<details><summary style="font-size: 20px">Provider Block Option</summary>
+
+[Provider Block Option](#example-of-provider-block) - you can create your **Terraform plan** by putting everything in one `main.tf` file, which then contains your Provider and variables settings explicitly in said plan.
+
+### Example of Provider Block:
+#### `main.tf`
+ ```hcl
+ # Configure the Azure Stack Provider
+ provider "azurestack" {
+  arm_endpoint    = "https://management.{region}.{domain}"
+  subscription_id = "xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx"
+  client_id       = "{applicationId}"
+  client_secret   = "{applicationPassword}"
+  tenant_id       = "xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx"
+}
+Rest of the file (...)
+```
+
+Official [Variables Guide](https://www.terraform.io/intro/getting-started/variables.html)
+
+</details>
+
 ## Creating a VM
 
-The examples that follow show how to create a VM using Terraform. The code changes depending on whether you are creating a VM with an unmanaged disk or a managed disk. Select the the type of disk you want to use to update the examples below with the appropriate code.
-
-<form id="diskType" onchange="result.value=name.value;result2.value=name.value">
-  <p>
-    <input type="radio" name="name" value='storage_os_disk {
-    name          = "myosdisk1"
-    vhd_uri       = "${azurestack_storage_account.test.primary_blob_endpoint}${azurestack_storage_container.test.name}/myosdisk1.vhd"
-    caching       = "ReadWrite"
-    create_option = "FromImage"
-  }' checked>An unmanaged disk
-  </p>
-  <div></div>
-  <p>
-    <input type="radio" name="name" value='storage_os_disk {
-    name              = "myosdisk1"
-    caching           = "ReadWrite"
-    create_option     = "FromImage"
-    managed_disk_type = "Standard_LRS"
-  }'>A managed disk
-  </p>
-</form>
-
-### Create a VM with Public IP template - using **Environment Variables** option
+The examples that follow show how to create VMs using Terraform. The code changes depending on whether you are creating a VM with an unmanaged or managed disk. You can select the the type of disk you want to use, which will then update the examples below with the appropriate code.
```

Comment:
follow should be follows

Reply:
I believe examples that follow is correct, would need be follows if it was "example"

Reply:
Yep, my bad